torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -282,13 +282,15 @@ Value* TracingState::getOutput(const IValue& iv, size_t i) {
 
     bool key_type_valid = key_type->isSubtypeOf(StringType::get()) ||
         key_type->isSubtypeOf(TensorType::get());
-    bool value_type_valid = value_type->isSubtypeOf(TensorType::get());
+    bool value_type_valid = value_type->isSubtypeOf(TensorType::get()) ||
+        value_type->isSubtypeOf(AnyTupleType::get());
```

Comment:
This means the value type can be any of tuple type, it might actually including something like `Tuple[int, int]`? In that case it won't be a valid input right? The valid type check here need to be more strict.

Reply:
Yeah, let's be strict on this check, we probably don't want to enable cases like Dict[str, Tuple[int, int]]

Reply:
sounds good. updated by checking the contained types of the tuple, and added a test condition. let me know!

Reply:
thanks, looks good to me.

@wanchaol, any more comment?